app/src/main/java/pl/futurecollars/invoicing/controller/InvoiceController.java

```diff
@@ -54,7 +57,11 @@
         if (buyerId != null) {
             invoicePredicate = invoicePredicate.and(invoice -> invoice.getTo().getId().equals(buyerId));
         }
-        return ResponseEntity.ok().body(invoiceService.filter(invoicePredicate));
+        if (invoicePredicate == null) {
+            return ResponseEntity.ok().body(invoiceService.getAll());
+        } else {
+            return ResponseEntity.ok().body(invoiceService.filter(invoicePredicate));
```

Comment:
Maybe you could try to generalize filter() method in service and there check if predicate is null and then do some operations. This way you could just leave ResponseEntity.ok().body(invoiceService.filter(invoicePredicate));

Reply:
My thought process was that if there are no query params then it will be faster to return all invoices with getAll() method than filter all invoices thorough my very generic Predicate(Objects::nonNull), but maybe there is a better way to do this